### DISCUSSION TITLE
Sending Images to remote dcm4chee running on a server using locally installed weasis as a desktop application over tls port 2762

### DISCUSSION THREAD

**Discussion by RajaMuneerBaigal** [nroduit/Weasis#519] [Q&A]
Hi,
I have set up DCM4CHEE on a public server using the following link using docker-compose:
https://github.com/dcm4che/dcm4chee-arc-light/wiki/Run-minimum-set-of-archive-services-on-a-single-host
I have installed Weasis on my locally running machine using the desktop application installer. The issue is I am having trouble when trying to export a Dicom image from a locally running Weasis using tls port 2762. I can export images to DCM4CHEE using http port without any issues but when using tls port seems like it shows null when try to export the image over port 2762. Below are the attached screenshots:

![image](https://github.com/nroduit/Weasis/assets/105891199/2391d717-9e02-48a3-82a0-88867b0ee0fa)
![image](https://github.com/nroduit/Weasis/assets/105891199/fcb3751e-21ef-42f3-a1b8-6ad4f7e4ea7f)
![image](https://github.com/nroduit/Weasis/assets/105891199/8017014a-cc23-41c2-8516-da00db1bdd54)
![image](https://github.com/nroduit/Weasis/assets/105891199/840ce9ab-8ee0-4e9b-9772-ff98d501a384)

I have also pointed my ip to the domain and tried putting the domain and even tried with the domain also uploading the fullchain.pem  to weasis keystore following the instructions from below link but I get a bad_certificate error in that case. Not sure what I am doing wrong or is it even possible to send dicom images to remote public server from locally running Weasis. Any help would be appreciated.

https://groups.google.com/g/dcm4che/c/wPOCR0czXeY/m/6iQjU0sXAAAJ

And with the domain I get the following error:

![image](https://github.com/nroduit/Weasis/assets/105891199/aeffe7ca-53d6-4536-8265-b6169db53155)

**Comment by nroduit** [comment]
The implementation was not ported to the client's GUI because, in my opinion, it's too complicated for a user to manage (having a keystore and a trust store, choosing the TLS version and ciphers).

The solution is to create a client in Keycloak and use [dicomweb](https://weasis.org/en/tutorials/dicomweb-config/i)(the documention for dcm4chee must be added). See this [post](https://groups.google.com/g/dcm4che/c/lvgFa-QEZRM/m/vWzI6wQqCgAJ).

**Reply by RajaMuneerBaigal** [reply]
I am running miminum services of dcm4chee, that are arc image, db and ldap only no keycloak or no secure ui image.

**Comment by nroduit** [comment]
> I am running miminum services of dcm4chee, that are arc image, db and ldap only no keycloak or no secure ui image.

So you can only connect in non-secure mode

**Reply by RajaMuneerBaigal** [reply]
How that's the question? What do I need to do ?  I tried directly connecting by putting the url in the web-node but I get bad_certificate errors

**Comment by nroduit** [comment]
It is not possible to use a secure protocol on the client side if on the server side the services are not secure.

It is already difficult for me to answer the various questions on the different projects I contribute. Unfortunately, I don't assist in configuring and securing applications. Only very specific questions and proven problems will be followed up.
